pkgs/test/nixpkgs-check-by-name/default.nix

```diff
@@ -25,14 +25,7 @@ let
         export NIX_LOG_DIR=$TEST_ROOT/var/log/nix
         export NIX_STATE_DIR=$TEST_ROOT/var/nix
         export NIX_STORE_DIR=$TEST_ROOT/store
-
-        # Ensure that even if tests run in parallel, we don't get an error
-        # We'd run into https://github.com/NixOS/nix/issues/2706 unless the store is initialised first
-        nix-store --init
```

Comment:
Shouldn't this still be necessary though? Nix shouldn't be affected by temp-env

Reply:
I don't think it is because mutexes ensure only one `nix` process is spawned at once, but the logic is complicated enough that it may be safer to leave it as-is.